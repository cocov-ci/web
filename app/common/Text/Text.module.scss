@import "styles/variables.module";

.text {
  strong {
    font-variation-settings: "wght" 600;
  }

  .title {
    font-size: $fontTitle;
    font-variation-settings: "wght" 400;
  }

  .description {
    font-size: $fontDescription;
    margin-bottom: 20px;
    white-space: pre-line;
  }
}
