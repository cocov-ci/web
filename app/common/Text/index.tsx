import classNames from 'classnames'
import React from 'react'

import styles from './Text.module.scss'

type TextProps = {
  children?: React.ReactNode
  className?: string
  title?: string
  variant?: 'title' | 'description'
}

const Text = ({ children, className, variant = 'title', title }: TextProps) => {
  return (
    <p
      className={classNames(styles.text, styles[variant], className)}
      title={title}
    >
      {children}
    </p>
  )
}

export default Text
