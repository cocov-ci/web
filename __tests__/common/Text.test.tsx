import { render } from '@testing-library/react'

import Text from 'app/common/Text'

import '@testing-library/jest-dom'

describe('Text', () => {
  it('renders Text', () => {
    const { container } = render(<Text />)

    expect(container.querySelector('p')).toBeVisible()
  })

  it('renders Text snapshots', () => {
    const { container } = render(<Text />)
    expect(container).toMatchSnapshot()
  })
})
