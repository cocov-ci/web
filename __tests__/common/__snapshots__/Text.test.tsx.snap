// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders Text snapshots 1`] = `
<div>
  <p
    class="title"
  />
</div>
`;
